### PR TITLE
feat: handlers for verifiable credentials

### PR DESCRIPTION
**Pull request by asasmita**
The context is authorization server for credential issuance, especially
when the authorization server is separate entity from credential issuer.

There are additional parameters that need to be processed or verified; such as c_nonce, issuer_state.
The approach taken here is to associate those parameters in the session, so it should be retrievable and output
when access token is introspected by credential issuer.

The PR introduce the following:
- a new type of client to indicate that the client is able to process/produce additional information in regards to credential issuance
- a new grant type for pre authorize code flow
- handler to process issuer_state at the /authorize endpoint that persist the parameter into session
- handler to generate c_nonce at /token endpoint and persist it into session for verification later on
- handler for pre authorize code flow at /token endpoint to exchange pre-authorize code and tx_code for access token (and refresh token)

## Checklist

- [X] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [X] I have read the [security policy](../security/policy).
- [X] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: handler/verifiable/flow_preauthcode_token.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	preAuthRequest, err := c.Storage.GetPreAuthorizeSession(ctx, signature, request.GetSession())
+	if err != nil && errors.Is(err, fosite.ErrNotFound) {
+		return errorsx.WithStack(fosite.ErrInvalidGrant.WithHint("Invalid Pre-Authorized Code or the Pre-Authorized Code has expired."))
+	} else if err != nil {
+		return errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebug(err.Error()))
+	}
+
+	// The authorization server MUST verify that the pre-authorization code is valid
+	// This needs to happen after store retrieval for the session to be hydrated properly
+	if err := c.AuthorizeCodeStrategy.ValidateAuthorizeCode(ctx, request, code); err != nil {
```
Comment (vivshankar) [inline handler/verifiable/flow_preauthcode_token.go:74]: Should we be introducing a new function specifically for pre-authorized code to make sure that the pre auth code isn't misused as the auth code?

**Review by vivshankar — CHANGES_REQUESTED** [review]
Only one comment. I think we should separate the storage interface for pre-auth code vs auth code.

File: handler/verifiable/flow_preauthcode_token.go (hunk 1 above)
Comment (asasmita) [reply handler/verifiable/flow_preauthcode_token.go:74]: yeah, make sense. Ok I'll separate it.

Comment (asasmita) [reply handler/verifiable/flow_preauthcode_token.go:74]: I'm reusing the generation and validation of the authorization code string only. Not the storing part. So, I think it should be ok.

**Review by vivshankar — APPROVED** [review]

**Pull request merged**